src/Control/Monad/Logic/Sequence/Internal.hs

```diff
@@ -154,32 +161,46 @@ single a = return (a :< mzero)
 
 instance Monad m => Functor (SeqT m) where
   {-# INLINEABLE fmap #-}
-  fmap f xs = xs >>= return . f
+  fmap f (SeqT q) = SeqT $ fmap (liftM (fmap f)) q
+  {-# INLINABLE (<$) #-}
+  x <$ SeqT q = SeqT $ fmap (liftM (x <$)) q
 
 instance Monad m => Applicative (SeqT m) where
   {-# INLINE pure #-}
-  {-# INLINE (<*>) #-}
+  {-# INLINABLE (<*>) #-}
   pure = fromView . single
-  (<*>) = liftM2 id
+  (<*>) = ap
+  (*>) = (>>)
+#if MIN_VERSION_base(4,10,0)
+  liftA2 = liftM2
+  {-# INLINABLE liftA2 #-}
+#endif
 
 instance Monad m => Alternative (SeqT m) where
   {-# INLINE empty #-}
   {-# INLINEABLE (<|>) #-}
   {-# SPECIALIZE INLINE (<|>) :: Seq a -> Seq a -> Seq a #-}
   empty = SeqT S.empty
-  (toView -> m) <|> n = fromView (m >>= \x -> case x of
-      Empty -> toView n
-      h :< t -> return (h :< cat t n))
+  m <|> n = fromView (altView m n)
+
+altView :: Monad m => SeqT m a -> SeqT m a -> m (View m a)
+altView (toView -> m) n = m >>= \x -> case x of
+  Empty -> toView n
+  h :< t -> return (h :< cat t n)
     where cat (SeqT l) (SeqT r) = SeqT (l S.>< r)
+{-# INLINE altView #-}
 
 instance Monad m => Monad (SeqT m) where
   {-# INLINE return #-}
   {-# INLINEABLE (>>=) #-}
   {-# SPECIALIZE INLINE (>>=) :: Seq a -> (a -> Seq b) -> Seq b #-}
   return = fromView . single
-  (toView -> m) >>= f = fromView (m >>= \x -> case x of
+  (toView -> m) >>= f = fromView $ m >>= \x -> case x of
+    Empty -> return Empty
+    h :< t -> f h `altView` (t >>= f)
+  (toView -> m) >> n = fromView $ m >>= \x -> case x of
```

Comment:
Please add ```{-# INLINEABLE (>>) #-}```

Reply:
maybe I'll make this change myself. After adding that pragma, the times are good again:


```
SeqT IO with msplit:
Batch generated in: 0.476231 seconds
Batch generated in: 0.456536 seconds
Batch generated in: 0.44283 seconds
Batch generated in: 0.449383 seconds
Batch generated in: 0.451309 seconds
Total time: 2.276441 seconds

SeqT Identity with msplit:
Batch generated in: 0.599356 seconds
Batch generated in: 0.579685 seconds
Batch generated in: 0.571126 seconds
Batch generated in: 0.558928 seconds
Batch generated in: 0.5614629999999999 seconds
Total time: 2.870699 seconds

SeqT Identity without msplit:
Batch generated in: 0.572662 seconds
Batch generated in: 0.574406 seconds
Batch generated in: 0.588789 seconds
Batch generated in: 0.502955 seconds
Batch generated in: 0.669292 seconds
Total time: 2.908248 seconds

```